image/verify/src/verifier.rs

```diff
@@ -586,7 +587,17 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
     // Check if SVN check is required
     fn svn_check_required(&mut self) -> bool {
         // If device is unprovisioned or if anti-rollback is disabled, don't check the SVN.
-        !(self.env.dev_lifecycle() == Lifecycle::Unprovisioned || self.env.anti_rollback_disable())
+        if cfi_launder(self.env.dev_lifecycle()) == Lifecycle::Unprovisioned
+            || cfi_launder(self.env.anti_rollback_disable())
+        {
```

Comment:
Should we add some non-deterministic delay here?

Reply:
We have multiple CFI delays explicitly added before starting firmware validation; not too inclined on adding a delay here.

Reply:
This can be addressed in another PR, but I think in general we want to the CFI double checks throughout the code-base to be done much later in time than the original check.